### ISSUE TITLE
DbtRuntimeError.__init__() missing 1 required positional argument: 'msg'

### ISSUE THREAD

**Issue by brendavarguez** [ClickHouse/dbt-clickhouse#169]
### Describe the bug

I am having this error message when trying to build both materialized tables and incremental tables. The funny thing is that sometimes I am able to build the tables and sometimes I get this error message.

```
DbtRuntimeError.__init__() missing 1 required positional argument: 'msg'
    
    > in macro materialization_table_clickhouse (macros/materializations/table.sql)
    > called by model stg_transaction_event_card (models/prd/staging/stg_transaction_event_card.sql)
```

When successful, the table is build with all the data. But when it fails, an empty table is created in my database.

#### For materialized tables
1. `profiles.yaml`
```
local_clickhouse:
      type: clickhouse
      schema: warehouse_dev
      user: "{{ env_var('CLICKHOUSE_USER') }}"
      password: "{{ env_var('CLICKHOUSE_PASS') }}"
      port: 9000
      host: localhost
      cluster: default
      retries: 1
      connect_timeout: 10
      send_receive_timeout: 3600 # 1hr
```
2. `stg_transaction_event_card.sql` example code:
```
{{
    config(
        materialized = "table",
        engine = "ReplicatedMergeTree",
        unique_key = "transaction_id",
        pre_hook = "TRUNCATE TABLE IF EXISTS {{ this }} ON CLUSTER default",
        indexes=[
          {'columns': ['transaction_id'], 'type': 'btree', 'unique': True },
          {'columns': ['last_mod'], 'type': 'btree'}
        ]
    )
}}

WITH txns_events AS (
    SELECT
        col1,
        col2,
        col3
    FROM postgresql('host:port', 'db_name', 'table_name', 'db_user', 'db_password', 'schema')
)

SELECT
    col1,
    col2,
    col3
FROM txns_events
```

#### For incremental tables
1. `profiles.yaml`
```
local_clickhouse:
      type: clickhouse
      schema: warehouse_dev
      user: "{{ env_var('CLICKHOUSE_USER') }}"
      password: "{{ env_var('CLICKHOUSE_PASS') }}"
      port: 9000
      host: localhost
      cluster: default
      retries: 1
      connect_timeout: 10
      send_receive_timeout: 3600 # 1hr
      use_lw_deletes: true
```

2. `stg_transaction_event_card.sql` example code:
```
{{
    config(
        materialized = "incremental",
        unique_key = "transaction_id",
        indexes=[
          {'columns': ['transaction_id'], 'type': 'btree', 'unique': True },
          {'columns': ['last_mod'], 'type': 'btree'}
        ]
    )
}}

WITH txns_events AS (
    SELECT
        col1,
        col2,
        col3
    FROM postgresql('host:port', 'db_name', 'table_name', 'db_user', 'db_password', 'schema')

    {% if is_incremental() %}

         AND (last_mod > (SELECT max(last_mod) FROM {{ this }}))

    {% endif %}
)

SELECT
    col1,
    col2,
    col3
FROM txns_events
```

#### Environment
* dbt version: `dbt-core=1.4.6`
* dbt-clickhouse version: `dbt-clickhouse=1.4.2`
* Python version: `Python=3.11.4`
* Operating system: M1 Mac

**Comment by genzgd** [comment]
It looks like there's a bug in error message handling/logging that keeps you from seeing the underlying ClickHouse exception  I've pushed a fix for that in the new release 1.4.4.  My guess is that's hiding the real error message which might be related to some Postgres issue.  If you can try the 1.4.4 dbt-clickhouse version that might point to the root cause.

You might also check your ClickHouse server logs for related errors.

**Comment by genzgd** [comment]
That looks like a network error in the underlying HTTP request from dbt to your ClickHouse server.  How long does this select usually take?  Where is the Clickhouse server you are connecting to?

**Comment by genzgd** [comment]
Oh, you're doing this on localhost, so it's probably an error between your localhost and the Postgres server?

**Comment by brendavarguez** [comment]
The query takes around 20 - 25min for `materialized=table`. For incremental it takes 1min and 30seconds approx. But I don't understand why there is an error between my localhost and Postgres when using `WITH` statements specifically. @genzgd

**Comment by genzgd** [comment]
It could just be a coincidence.  Unfortunately the error still doesn't provide much information once it's been processed through several DBT layers.   You could try running the same query using just the underlying python client (`clickhouse-driver` if you are using native protocol over port 9000, `clickhouse-connect` if you are using http over port 8123).  The error is currently coming directly from the `clickhouse-driver` query so you should see the same thing with hopefully more information.

Does the DBT debug log show anything in addition to that error message?

**Comment by brendavarguez** [comment]
Found my issue!

- So, basically it wasn't an error on the network, it was a memory issue. Scaling my memory did the work!
- Other people also suggested updating send and receive timeout as well as `max_insert_threads`.

**Comment by brendavarguez** [comment]
Found my issue!

- So, basically it wasn't an error on the network, it was a memory issue. Scaling my memory did the work!
- Other people also suggested updating send and receive timeout as well as `max_insert_threads`.
- In addition, something I discovered is that, it doesn't matter if my CTE has filters (such as `date >= 'YYYY-MM-DD')`, clickhouse will load all the data. Hopefully this is useful for someone to check if they are also running out of memory.
